### PR TITLE
Add more logging to CallService_OtlpGrpcEndPoint_ExternalFile_FileChanged_UseConfiguredKey

### PR DESCRIPTION
**Pull request by JamesNK**
Addresses https://github.com/dotnet/aspire/issues/4637

I don't know what could cause this issue. I've added more logging to help debug future failures.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/4638)

### CODE REVIEW

File: src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs [hunk 1]
```diff
@@ -2,21 +2,30 @@
...
+
+    public PostConfigureDashboardOptions(IConfiguration configuration, ILogger<PostConfigureDashboardOptions> logger)
     {
         _configuration = configuration;
+        _logger = logger;
     }
 
     public void PostConfigure(string? name, DashboardOptions options)
     {
+        _logger.LogInformation($"PostConfigure {nameof(DashboardOptions)} with name '{name}'.");
```
Comment (radical) [inline src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs:27]: Should this be `LogDebug`?

File: src/Aspire.Dashboard/Configuration/ValidateDashboardOptions.cs [hunk 2]
```diff
@@ -2,14 +2,28 @@
...
+    }
+
+    public ValidateDashboardOptions(ILogger<ValidateDashboardOptions> logger)
+    {
+        _logger = logger;
+    }
+
     public ValidateOptionsResult Validate(string? name, DashboardOptions options)
     {
+        _logger.LogInformation($"Validating {nameof(DashboardOptions)}.");
```
Comment (radical) [inline src/Aspire.Dashboard/Configuration/ValidateDashboardOptions.cs:25]: Should this be `LogDebug` too?

File: src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs (hunk 1 above)
Comment (JamesNK) [inline src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs:27]:
```suggestion
        _logger.LogDebug($"PostConfigure {nameof(DashboardOptions)} with name '{name}'.");
```

File: src/Aspire.Dashboard/Configuration/ValidateDashboardOptions.cs (hunk 2 above)
Comment (JamesNK) [inline src/Aspire.Dashboard/Configuration/ValidateDashboardOptions.cs:25]:
```suggestion
        _logger.LogDebug($"Validating {nameof(DashboardOptions)}.");
```

**Review by radical — APPROVED** [review]

**Pull request merged**